.github/workflows/linting.yml

```diff
@@ -4,9 +4,7 @@ on:
   - workflow_dispatch
 permissions:  # added using https://github.com/step-security/secure-workflows
   contents: read
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+
```

Comment:
The following will ensure that only Pull Request events are cancelled by concurrency, but not all Push events. WDYT?

```suggestion
concurrency:
  group: format('{0}-{1}', ${{ github.workflow }}-${{ github.ref }}, ${{ github.head_ref || github.run_id }})
  cancel-in-progress: true
```

Reply:
This is the approach described in the GitHub documentation.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-fallback-value

Reply:
And we are testing in the following repositories and have confirmed that it is canceled on Pull Request but not on Push.
https://github.com/ydah/concurrency-test/actions
![concurrency](https://user-images.githubusercontent.com/13041216/211561309-51b46d68-2a98-4d6e-931e-a63fe8b5715d.png)

Reply:
> The following will ensure that only Pull Request events are cancelled by concurrency, but not all Push events.

I like the idea. But `github.ref` looks unnecessary. `github.head_ref` if PR, otherwise `github.run_id` suffices. I've updated to the below:

```ruby
group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context